components/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect } from 'react'
+import { useQuery } from '@apollo/client'
+import gql from 'graphql-tag'
+import styled from 'styled-components'
+import SportClassCard from './SportClassCard'
+import getWeek from '../../lib/getWeek'
+
+export const ALL_CLASSES_QUERY = gql`
+  query ALL_CLASSES_QUERY {
+    allSportClasses {
+      id
+      name
+      freeSpots
+      status
+      week
+      day
+      startTime
+      teacher
+      duration
+    }
+  }
+`
+
+export default function Schedule() {
+  let { data, error, loading } = useQuery(ALL_CLASSES_QUERY)
+  if (loading) return <p>Loading...</p>
+  if (error) return <p>Error: {error.message}</p>
+
+  const [week, setWeek] = useState(0)
+  //todo: adjust dependency to update when pagination buttons are hit
+  useEffect(() => {
+    setWeek(getWeek())
+  }, [])
+
+  const FilterClassToDay = (dayName) => {
+    return data.allSportClasses
+      .filter(
+        (sportClass) =>
+          (sportClass.week === week && sportClass.day) === dayName,
+      )
+      .map((sportClass) => <SportClassCard classData={sportClass} />)
+  }
+
+  return (
+    <>
+      <div>{getWeek()}</div>
+      {/* todo: add functionality to buttons */}
+      <button>Previous Week</button>
+      <button>Next Week</button>
+      <TableContainer>
+        <TableHeader>
+          <tr>
+            <th>Monday</th>
+            <th>Tuesday</th>
+            <th>Wednesday</th>
+            <th>Thursday</th>
+            <th>Friday</th>
+            <th>Saturday</th>
+            <th>Sunday</th>
+          </tr>
+        </TableHeader>
+        <tbody>
+          <TableRow>
+            <TableField>{FilterClassToDay('Monday')}</TableField>
```

Comment:
Add the names of the days in an array or extract it so you don't have to repeat all that code. 
You're using the exact same values in the `<TableHeader>` and here. Make it easier for yourself by adding these values in an array and loop through it.

The additional benefit is that when you have translation, you can easily swap out the array for a different one without changing the code

Reply:
I see that you are using the name days in multiple files. I would suggest abstracting it to a separate file (`.ts`) where you export it and import it where you need it.

source:  https://stackoverflow.com/questions/24998624/day-name-from-date-in-js/24998705

```javascript
// All lowercase; if you need uppercase you can do it with either css (text-transform: capitalize) or use javascript
const days = ['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday'];

const d = new Date(dateString);

// Have the full name
export const dayName = days[d.getDay()];

// Only the first 3 letters
export const shortdayName = d.toString().split(' ')[0];
```

Now you have multiple options; also when it comes to translating, you could easily swap out the array